src/renderer.js

```diff
@@ -516,58 +507,44 @@ class gltfRenderer
         }
     }
 
-    initializeEnvironment(gltf, scene)
+    initializeEnvironment(environment, scene)
     {
         scene.envData = {};
 
-        if (scene !== undefined && scene.imageBasedLight !== undefined)
-        {
-            const diffuseTextureIndex = scene.imageBasedLight.diffuseEnvironmentTexture;
-            const specularTextureIndex = scene.imageBasedLight.specularEnvironmentTexture;
-            const sheenCubeMapIndex = scene.imageBasedLight.sheenEnvironmentTexture;
+        const diffuseTextureIndex = environment.textures.length - 6;
+        const specularTextureIndex = environment.textures.length - 5;
+        const sheenTextureIndex = environment.textures.length - 4;
 
```

Comment:
Add check if environment is valid. Otherwise, index might become negative and the program might need to be restarted.